src/Microsoft.Identity.Client/AppConfig/ApplicationOptions.cs

```diff
@@ -28,48 +28,86 @@
 namespace Microsoft.Identity.Client.AppConfig
 {
     /// <summary>
-    ///     Options object with string values loadable from JSON configuration (as in an asp.net configuration scenario)
+    /// Options object with string values loadable from a configuration file 
+    /// (for instance a JSon file as in an asp.net configuration scenario)
+    /// See https://aka.ms/msal-net-application-configuration
     /// </summary>
     public abstract class ApplicationOptions
     {
         /// <summary>
+        /// Client ID (also known as App ID) of the application as registered in the
+        /// application registration portal (https://aka.ms/msal-net-register-app)
         /// </summary>
         public string ClientId { get; set; }
 
         /// <summary>
-        /// Can be domain, can be guid tenant, can be meta-tenant (e.g. consumers).
+        /// Tenant from which the application will allow users to sign it. This can be:
+        /// a domain associated with a tenant, a guid (tenant id), or a meta-tenant (e.g. consumers).
+        /// This property is mutually exclusive with <see cref="AadAuthorityAudience"/>
         /// </summary>
+        /// <remarks>The name of the property was chosen o ensure compatibility with AzureAdOptions 
+        /// in ASP.NET Core configuration files (even the semantics would be tenant)</remarks>
         public string TenantId { get; set; }
 
         /// <summary>
-        /// Mutually exclusive with TenantId...
+        /// Sign-in audience. This property is mutually exclusive with TenantId...
```

Comment:
The fact that is mutually exclusive does not help the developer. It would be more helpful to say smth like "if you specify the TenantId, this is ignored" (or vice-versa)

Reply:
I agree @bogavril@microsoft.com 

---
In reply to: [248341744](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/790#discussion_r248341744) [](ancestors = 248341744)

Reply:
If you specify both, we should fail in the builders and not pick one.  That's why the original language was mutually exclusive.  But yes we can/should talk here on desired behavior.

---
In reply to: [248321086](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/790#discussion_r248321086) [](ancestors = 248321086)

Reply:
@bgavrilMS @jennyf19 I discussed with @MarkZuber today, and I like the fact that what he's proposing leads to un-ambiguous behavior. There is no notion of fallback. if you add 2 mutually exclusive properties, we'll throw with a meaningful exception.